### PR TITLE
Small renamings in generate_bids_dataset

### PR DESCRIPTION
**Pull request by SylvainTakerkart**
This corresponds to some of what we discussed and decided:
- `generate_bids_dataset`: rename data_file -> data_source, check for consistency 
- remove `session`, use `data_instance` / `ephys_instance`  or similar instead 

Not yet ready for review, coz I need to check for consistency, ahah ;)

### CODE REVIEW

**Comment by SylvainTakerkart** [comment]
OK, I've looked around, it seems ok to me! There will be other things that will need to evolve (the docstring for the BEP032Data class, the list of ESSENTIAL_CSV_COLUMNS), but I don't want to touch them yet... So this PR remains very small and symbolic!

**Comment by JuliaSprenger** [comment]
And since we are on it, maybe also `register_data_files` should rather be called `register_data_sources`? What do you think?

**Comment by SylvainTakerkart** [comment]
thanks! I caught a few other spots that needed to be corrected...

WDYT?

**Review by JuliaSprenger — APPROVED** [review]

File: bep032tools/generator/tests/test_BEP032Generator.py [hunk 1]
```diff
@@ -94,8 +94,8 @@ def test_data_files_complex(self):
         tasks = ['task1', 'task2']
         for run in runs:
             for task in tasks:
-                self.bep032tools_data.register_data_files(*nix_files,
-                                                   run=run, task=task)
+                self.bep032tools_data.register_data_sources(*nix_files,
+                                                            run=run, task=task)
```
Comment (JuliaSprenger) [inline bep032tools/generator/tests/test_BEP032Generator.py:98]:
```suggestion
                self.bep032tools_data.register_data_sources(*nix_files, run=run, task=task)
```

**Pull request merged**